### ISSUE TITLE
OperaDriver tries to extract launcher binaries from wrong resource path in OperaLaunchers JAR

### ISSUE THREAD

**Issue by benjaminhawkeslewis** [operasoftware/operaprestodriver#39]
In OperaLauncherRunnerSettings, OperaDriver 0.8.1 (downloaded from Maven) tries to extract launcher binaries from the OperaLaunchers JAR. But it tries to extract them from a non-existent "launchers" directory inside the JAR.

``` java
    URL
        sourceLauncher =
        OperaLaunchers.class.getClassLoader().getResource("launchers/" + launcherName);
```

I believe this line would resolve correctly:

``` java
OperaLaunchers.class.getClassLoader().getResource(launcherName);
```

It appears that when built with Ant, OperaDriver builds the launchers into the OperaDriver JAR itself.

Does this problem reflect a difference between the ZIP and Maven packaging?

**Comment by rjatkins** [comment]
I'm also seeing this issue. The operalaunchers-0.3.jar in maven has the launchers packaged in the top level directory of the jar, whereas building the jar through ant packages the launchers into the operadriver jar in the launchers directory.

The solution above works for me. Another alternative is simply to rebuild and redeploy the operalaunchers jar to use the launchers directory. Either way, you'll need to deploy another version of operadriver to maven to pick up the changes.

**Comment by scrobbleme** [comment]
Duplicate, which was just solved? https://github.com/operasoftware/operadriver/pull/41

**Comment by rjatkins** [comment]
Agreed. #41 was my attempt at a fix for this issue, but the fix was done in operalaunchers instead (which works just as well). This can now be closed as fixed in OperaDriver 0.9.

**Comment by andreastt** [comment]
Thanks for verifying that, rjatkins!
